### PR TITLE
Refactor products_controller move #find_order to before_action and ad…

### PR DESCRIPTION
**Pull request by tanmay3011**
…d specs for find order
- Avoid raising exception #find_product rather handle it explicitly.
- Move find_product to before_action in ProductsController
- Add specs for #find_product
- ReIndent all private methods in ProductsController to fix Hound issues related to indentation.
- Segregate #find_product and #authorize_product! to avoid double render error.

### CODE REVIEW

File: api/app/controllers/spree/api/v1/products_controller.rb [hunk 1]
```diff
@@ -98,6 +96,10 @@ def product_params
             params.require(:product).permit(permitted_product_attributes)
           end
 
+          def find_product
```
Comment (tanmay3011) [inline api/app/controllers/spree/api/v1/products_controller.rb]: All private methods are indented with 2 extra spaces so just maintaining the consistency.

Comment (damianlegawiec) [inline api/app/controllers/spree/api/v1/products_controller.rb]: @tanmay3011 private/protected methods shouldn't be indented more than any other methods, let's keep to rubocop's rules :)

File: api/app/controllers/spree/api/v1/products_controller.rb [hunk 2]
```diff
@@ -87,39 +86,43 @@ def update
...
+        def find_product
+          super(params[:id])
+        end
 
-          def option_types_params
-            params[:product].fetch(:option_types, [])
+        def variants_params
+          variants_key = if params[:product].has_key? :variants
+            :variants
+          else
```
Comment (tanmay3011) [inline api/app/controllers/spree/api/v1/products_controller.rb]: No need of else if

**Comment by tanmay3011** [comment]
@damianlegawiec Fixed indentation throughout the controller.

**Comment by priyank-gupta** [comment]
:+1: 
@damianlegawiec @Mafi88 ?

**Comment by damianlegawiec** [comment]
@tanmay3011 can you provide a more descriptive description? Lots of changes :)

**Comment by tanmay3011** [comment]
@damianlegawiec Added! Missed it since it was a 2weeks old PR. :p

**Comment by bbonislawski** [comment]
I've rebased it and going to merge it from #8587

**Pull request closed without merging**